### ISSUE TITLE
Can't merge for multiple python versions

### ISSUE THREAD

**Issue by pedrocasi** [dashingsoft/pyarmor#1873]
I'm trying to obfuscate for multiple python versions as described [here](https://pyarmor.readthedocs.io/en/latest/tutorial/advanced.html#obfuscating-scripts-for-multiple-python-versions).

I have folders 3.11 and 3.12 containing code obfuscated with the related python versions.  Each one contains a runtime package folder pyarmor_runtime_00xxxx (xxxx is my  registration code). When I run command

`python3 -m pyarmor.cli.merge --output multiversion 3.12 3.11`

I get the following error
INFO     start to merge '3.12', '3.11'...
INFO     scan runtime package in the path "3.11"
ERROR    **no runtime package found**

If instead I run 

`python3 -m pyarmor.cli.merge --runtime-name pyarmor_runtime_00xxx --output multiversion 3.12 3.11`

the merge operation completes but folder multiversion/pyarmor_runtime_00xxx only contains the __init__.py, file pyarmor_runtime.so is missing. And this means a runtime error: **ModuleNotFoundError: No module named 'pyarmor_runtime_004652.pyarmor_runtime'**

Alternatively, I also tried

`python3 -m pyarmor.cli.merge --no-runtime --output multiversion 3.12 3.11`

In this case, the merge operation completes and folder multiversion/pyarmor_runtime_00xxx seems ok but it still triggers **RuntimeError: unauthorized use of script (1:1102)**. If I replace folder multiversion/pyarmor_runtime_00xxx with 3.12/pyarmor_runtime_00xxx everything works. Needless to say, my runtime python version is 3.12.

Any suggestion?

**Comment by jondy** [comment]
Which pyarmor version are you used? Try to upgrade to latest pyarmor

**Comment by jondy** [comment]
Also check `pyarmor cfg plugins`, make sure there is one and only one `MultiplePythonPlugin` in each Python version

**Comment by pedrocasi** [comment]
Thank you for your quick reply. The pyarmnor version is 8.5.10 which is the latest available on PyPi. And I do have the `MultiplePythonPlugin` on the cfg:

```
(..)
Local settings
  plugins = CodesignPlugin DarwinUniversalPlugin MultiPythonPlugin
```

**Comment by jondy** [comment]
Please check the content of path `3.12/pyarmor_runtime_00xxxx` and `3.11/pyarmor_runtime_00xxxx`?

After merged, check the content `multipleversion/pyarmor_runtime_00xxxx`

I suggest first remove whole `3.12`, `3.11` and `multipleversion`, enable debug option `-d` to check output log

`pyarmor/cli/merge.py` is open source, it's just merge two folder files.

**Comment by pedrocasi** [comment]
I took a peek at function merge_runtimes inside pyarmor/cli/merge.py. It copies the __init__.py of the first runtime package and all the subfolders on every runtime package. It does nothing about the pyarmor_runtime.so files and so the merged folder won't have such a file. Isn't it necessary for runtime decoding?  I'm guessing this isn't the expected behaviour.

**Comment by jondy** [comment]
It's expected behaviours.

If plugin "MultiplePythonPlugin" works and no other user's plugin, the runtime package should be one subfolder in the output path after `pyarmor gen ...` 

For example, if output path is `dist`, it will be `dist/pyarmor_runtime_00xxxx/py312/`

So please check the output path of each Python version

**Comment by lozanosky** [comment]
If anyone have come here with this problem, the solution for me was:

1 - Be sure MultiPythonPlugin appears only one time in each python version you want to include in your merged package, in my case python 3.7.7 and python 3.10.11:

```
INFO     Python 3.7.7
INFO     Pyarmor 9.1.0 (trial), 000000, non-profits
INFO     Platform windows.x86_64

------------------------------------------------------------
Section: builder

Current settings
  plugins = CodesignPlugin DarwinUniversalPlugin MultiPythonPlugin

Global settings

Local settings
  plugins = CodesignPlugin DarwinUniversalPlugin MultiPythonPlugin
```
```
INFO     Python 3.10.11
INFO     Pyarmor 9.1.0 (trial), 000000, non-profits
INFO     Platform windows.x86_64

------------------------------------------------------------
Section: builder

Current settings
  plugins = CodesignPlugin DarwinUniversalPlugin MultiPythonPlugin

Global settings

Local settings
  plugins = CodesignPlugin DarwinUniversalPlugin MultiPythonPlugin
```
2 - Once this setting is ok, be sure you have added the corresponding pyarmor version in your pythonpath (don't obfuscate the code with python 3.10 if you have the pyarmor package version for python 3.7 in your pythonpath, be consistent with the versions)

3 - Once this is also done, obfuscate your packages:
(with pyarmor for python 3.7 in my pythonpath)
"my_path_to\python\3.7.7\python.exe" -m pyarmor.cli gen -r -i -O L:\temp\crypted\result\3.7.7 O:\temp\crypted\moduleTest

(with pyarmor for python 3.10 in my pythonpath)
"my_path_to\python\3.10.11\python.exe" -m pyarmor.cli gen -r -i -O L:\temp\crypted\result\3.10.11 O:\temp\crypted\moduleTest

4 - Merge with the lower pyarmor packaged for lower python version (pyarmor for python 3.7 in my python path, and mergin with python 3.7 in my case):

"my_path_to\python\3.7.7\python.exe" -m pyarmor.cli.merge -O L:\temp\crypted\result\merged L:\temp\crypted\result\3.7.7\moduleTest L:\temp\crypted\result\3.10.11\moduleTest
